modules/http/src/test/java/org/mule/module/http/functional/requester/HttpRequestOutboundAttachmentsTestCase.java

```diff
@@ -157,6 +165,81 @@ public void sendingAttachmentBiggerThanAsyncWriteQueueSizeWorksOverHttps() throw
         assertThat((Integer) response.getMessage().getInboundProperty(HTTP_STATUS_PROPERTY), equalTo(OK.getStatusCode()));
     }
 
+    @Test
+    public void canSetCustomAttachmentHeaders() throws Exception {
+        String attachmentName = canSetCommonSetup();
+
+        Part resultPart = getPart(attachmentName);
+        assertThat(resultPart.getHeader("a-custom-header"), equalTo("custom"));
+    }
+
+    @Test
+    public void canSetContentTransferEncodingHeader() throws Exception {
+        String attachmentName = canSetCommonSetup();
+
+        Part resultPart = getPart(attachmentName);
+        assertThat(resultPart.getHeaders(CONTENT_TRANSFER_ENCODING.toLowerCase()).size(), is(1));
+        assertThat(resultPart.getHeader(CONTENT_TRANSFER_ENCODING.toLowerCase()), equalTo("base64"));
+    }
+
+    @Test
+    public void canSetContentDispositionHeader() throws Exception {
+        String attachmentName = canSetCommonSetup();
+
+        Part resultPart = getPart(attachmentName);
+        assertThat(resultPart.getHeaders(CONTENT_DISPOSITION.toLowerCase()).size(), is(1));
+        assertThat(resultPart.getHeader(CONTENT_DISPOSITION.toLowerCase()), equalTo("form-data; name=\"attachment1\"; documentId=1"));
+    }
+
+    @Test
+    public void canSetContentIdHeader() throws Exception {
+        String attachmentName = canSetCommonSetup();
+
+        Part resultPart = getPart(attachmentName);
+        assertThat(resultPart.getHeaders(CONTENT_ID.toLowerCase()).size(), is(1));
+        assertThat(resultPart.getHeader(CONTENT_ID.toLowerCase()), equalTo("my-id"));
+    }
+
+    @Test
+    public void canSetContentTypeHeader() throws Exception {
+        String attachmentName = canSetCommonSetup();
```

Comment:
Do everything in one test if nothing changes from one to another, otherwise you are making 5 requests when 1 will do.